alternatives.md

```diff
@@ -0,0 +1,129 @@
+dependencies, etc.
+
+Babelfish provides a unified IR across languages, which is defined
+as a non-goal of Kythe.
+
+At the same time since Babelfish only processes one file at the time, it cannot
+resolve symbolic references in the code, while the goal on Kythe project is to
+provide this information with a very high accuracy.
+
+Kythe requires to instrument language compilers and build systems (by modifying
+the sources of compiler and build system), while Babelfish uses native language
+parsers to get an AST, allowing to develop a new language driver in less time.
+
+Kythe requires a project to be built to be processed because of integration with
+compilers and build systems, while Babelfish can process only individual files.
+
+## [Language Server Protocol](https://microsoft.github.io/language-server-protocol/)
+
+> A Language Server is meant to provide the language-specific smarts and communicate with development tools over a protocol that enables inter-process communication.
+
+> The idea behind the Language Server Protocol (LSP) is to standardize the protocol for how such servers and development tools communicate. This way, a single Language Server can be re-used in multiple development tools, which in turn can support multiple languages with minimal effort.
+
+LSP defines a common protocol and RPC for queries like Go-To-Definition,
+Usages, etc. But it does not define a common representation of AST because
+the goal of the project is to enable easy access to analysis that is done
+by compilers and language SDKs.
+
+Babelfish provides a common representation for ASTs allowing to use its output
+for static analysis by other tools. Babelfish cannot resolve symbolic references,
+thus it cannot be used to query usages.
+
+## [srclib](https://srclib.org/)
+
+> srclib makes developer tools like code search and static analyzers better. It supports things like jump to definition, find usages, type inference, and documentation generation.
+
+> srclib handles: package detection, global dependency resolution, type inference, querying the graph of definitions and references in code, versioning using different VCS systems, and semantic blaming.
+
+<!-- TODO -->
+
+## [ctags](http://ctags.sourceforge.net/)
+
+> Ctags generates an index (or tag) file of language objects found in source files that allows these items to be quickly and easily located by a text editor or other utility. A tag signifies a language object for which an index entry is available (or, alternatively, the index entry created for that object).
+
+Both Babelfish and Ctags provide positional information for identifiers,
+classes, directives, etc.
+
+The main difference is that Ctags does not provide any form of AST, while
+Babelfish provides native language AST as well as language-independent UAST.
+
+Also, Babelfish ecosystem allows making more complex queries over AST.
+
+## [ANTLR](https://github.com/antlr/antlr4)
```

Comment:
Should this section be about ANTLR and other parser generators? Or include a mention about this being applicable to comparison with other parser generators?